### DISCUSSION TITLE
How to use multi CDN domains better in Hiddify چگونه چندین دامنه CDN را در هیدیفای بهتر مدیریت کنیم

### DISCUSSION THREAD

**Discussion by lymanjre** [hiddify/Hiddify-Manager#332] [Q&A سوالات رایج] [answered]
هنگامی که چندین دامنه مخصوصا دامنه CDN داریم و کاربران ما نیز در شبکه های مختلف همراه اول؛ ایرانسل؛ مخابرات و .. هستند؛ چگونه می توان با استفاده از قابلیت های پنل هیدیفای آن ها را بهتر مدیریت کرد و کانفیگ های بهتری ساخت؟

**Comment by lymanjre** [comment, marked as answer]
# روش پیشنهادی اول
برای این کار پیشنهاد می گردد
چندین ساب دامین CDN مختلف برای پنل خود در کلادفلر تعریف کنید. دفت کنید که باید تیک پروکسی فعال باشد. [نحوه تعریف دامنه](https://github.com/hiddify/hiddify-config/wiki/%D8%A7%D9%86%D9%88%D8%A7%D8%B9-%D8%AF%D8%A7%D9%85%D9%86%D9%87-%D9%88-%D9%86%D8%AD%D9%88%D9%87-%D8%AB%D8%A8%D8%AA-%E2%80%8C%D8%A2%D9%86%E2%80%8C%D9%87%D8%A7)

مثلا:

sub1.domain.com  (proxy on)

sub2.doamin.com (proxy on)

sub3.domain.com (proxy on)


بعد از آن در شبکه های مختلف آیپی های تمیز کلادفلر را پیدا می کنیم. یعنی روی کلاینتی که به شبکه همراه اول متصل است؛ آیپی تمیز کلادفلر را پیدا می کنیم. اگر با روش پیدا کردن آیپی تمیز گفته شده آشنا نیستید؛ [اینجا](https://github.com/hiddify/hiddify-config/wiki/%DA%86%DA%AF%D9%88%D9%86%DA%AF%DB%8C-%DB%8C%D8%A7%D9%81%D8%AA%D9%86-%D8%A2%DB%8C%D9%BE%DB%8C-%D8%AA%D9%85%DB%8C%D8%B2-%DA%A9%D9%84%D8%A7%D8%AF%D9%81%D9%84%D8%B1) را کلیک کنید.
 برای ایرانسل؛ مخابرات و... نیز به همین شکل ادامه می دهیم. در نهایت شما چند آیپی تمیز کلادفلر به ازاء هر اپراتورها  دارید. 

حالا مجدد برای آیپی های پیدا شده؛ ساب دامین معمولی می سازیم. یعنی تیک پروکسی خاموش باشد.

مثلا:

mci.domain.com  (proxy off)  ==>  آیپی تمیز کلادفلر در شبکه همراه اول

irc.domain.com   (proxy off)  ==>  آیپی تمیز کلادفلر در شبکه ایرانسل

mkb.domain.com  (proxy off)  ==>  آیپی تمیز کلادفلر در شبکه مخابرات


در نهایت در پنل هیدیفای به منوی دامنه ها بروید. برای هر کدام از رکوردهای ثبت شده یک دامنه ایجاد کنید.

#### برای همراه اول

![Screenshot_20230306_084329](https://user-images.githubusercontent.com/125398461/223029768-d9817136-fd07-413d-a273-ceffd8ddc328.png)

نکته بسیار مهم:

- در فیلد **نمایش کانفیگ های دامنه** می توانید دامنه های CDN دیگر را از لیست حذف نمایید. در این حالت کانفیگ های همین دامنه فقط نمایش داده می شود . باعث می شود آیپی تمیز ثبت شده به درستی کار کند.

#### برای ایرانسل

![Screenshot_20230306_084411](https://user-images.githubusercontent.com/125398461/223029873-be4a8617-95c2-4c93-befd-7bbf50b91073.png)

نکته بسیار مهم:

- در فیلد **نمایش کانفیگ های دامنه** می توانید دامنه های CDN دیگر را از لیست حذف نمایید. در این حالت کانفیگ های همین دامنه فقط نمایش داده می شود . باعث می شود آیپی تمیز ثبت شده به درستی کار کند.

#### برای شبکه مخابرات 

![Screenshot_20230306_084440](https://user-images.githubusercontent.com/125398461/223029865-7625d50d-3c03-43d1-b191-e6c5f2fe00d4.png)

نکته بسیار مهم:

- در فیلد **نمایش کانفیگ های دامنه** می توانید دامنه های CDN دیگر را از لیست حذف نمایید. در این حالت کانفیگ های همین دامنه فقط نمایش داده می شود . باعث می شود آیپی تمیز ثبت شده به درستی کار کند.


برای سایر شبکه ها مثل شاتل؛ آسیاتک و ... همین الگو را ادامه دهید.

**Reply by lymanjre** [reply]
# روش پیشنهادی دوم:

در این روش به ۲ دامنه نیاز داریم  که دامنه اول را طبق روش قبل استفاده می کنیم یعنی چندین ساب دامین CDN مختلف برای پنل خود در کلادفلر تعریف کنید. دفت کنید که باید تیک پروکسی فعال باشد. [نحوه تعریف دامنه](https://github.com/hiddify/hiddify-config/wiki/%D8%A7%D9%86%D9%88%D8%A7%D8%B9-%D8%AF%D8%A7%D9%85%D9%86%D9%87-%D9%88-%D9%86%D8%AD%D9%88%D9%87-%D8%AB%D8%A8%D8%AA-%E2%80%8C%D8%A2%D9%86%E2%80%8C%D9%87%D8%A7)

مثلا:

sub1.domain1.com (proxy on)

sub2.doamin1.com (proxy on)

sub3.domain1.com (proxy on)


بعد برای آیپی های تمیز هم به همان روش بالا جلو برید. مثلا:

mci.domain1.com (proxy off) ==> آیپی تمیز کلادفلر در شبکه همراه اول

irc.domain1.com (proxy off) ==> آیپی تمیز کلادفلر در شبکه ایرانسل

mkb.domain1.com (proxy off) ==> آیپی تمیز کلادفلر در شبکه مخابرات


در نهایت اینجا یه تفاوت داریم با روش قبل. میایم با استفاده از دامین دوم به ازاء هر رکورد از دامین اول یک رکورد cname  تعریف می کنیم. این کار برای جلوگیری از لو رفتن و فیلتر شدن دامنه و ساب دامنه های اول می باشد. یعنی مثلا:

sub1.domain2.com ==> sub1.domain1.com (proxy off)

sub2.domain2.com ==> sub2.domain1.com (proxy off)

sub3.domain2.com ==> sub3.domain1.com (proxy off)

mci.domain2.com  ==> mci.domain1.com (proxy off)

irc.domain2.com ==> irc.domain1.com (proxy off)

mkb.domain2.com ==> mkb.domain1.com (proxy off)

در نهایت در پنل هیدیفای به منوی دامنه‌ها بروید. برای هر کدام از رکوردهای ثبت شده یک دامنه ایجاد کنید.

مثلا برای **همراه اول**

Domain: sub1.domain2.com

Type: CDN

Force CDN host:  mci.domain2.com


یا برای **ایرانسل**

Domain: sub2.domain2.com

Type: CDN

Force CDN host:  irc.domain2.com


یا برای **شبکه مخابرات**

Domain: sub3.domain2.com

Type: CDN

Force CDN host:  mkb.domain2.com

**نکته بسیار مهم:**

- در فیلد نمایش کانفیگ های دامنه می توانید دامنه های CDN دیگر را از لیست حذف نمایید. در این حالت کانفیگ های همین دامنه فقط نمایش داده می شود . باعث می شود آیپی تمیز ثبت شده به درستی کار کند.

**و توجه داشته باشید:**

- تمام این روشها راه حل های ابتکاری هستند و شما می توانید با دانش و خلاقیت خودتون تغییر بدین.

**Reply by Hazily1536** [reply]
"نکته بسیار مهم" گنگ هست. لطفا بیشتر توضیح بدید. تیک ها که همه ورداشته ست!

**Reply by lymanjre** [reply]
ببینید هنگام ایجاد دامنه در پنل هیدیفای یک فیلد وجود داره با عنوان 
**نمایش کانفیگ های دامنه**

می توانید برای دامنه ای که در حال ثبتش هستید تعیین کنید کانفیگ های مربوط به کدام دامنه های دیگر را نمایش دهد. 
یعنی مثلا شما چند دامنه دیگر قبلا وبت کرده‌اید که روی آنها کانکشن پروکسی وجود دارد. در این حالت عملا شما یک دامنه دیگر ایجاد می‌کنید که روی آن کانفیگ وجود ندارد بلکه متصل به کانفیگهای دامنه های قبلی‌ می‌شود. این دامنه هیچوقت فیلتر نمی شود چون کانفیگ پروکسی روی آن وجود ندارد. بنابراین کاربرها به راحتی می توانند آپدیت کانکشن‌ها را دریافت کنند.

**Reply by shuvvan** [reply]
تشکر بابت آموزش ولی آیا همه با یه کانفیگ وصل میشن یابرای هر نوع اپراتور باید کانفیگ جدا بدیم

**Reply by lymanjre** [reply]
برای هر اپراتور کانفیگ جدا بدین

**Reply by shuvvan** [reply]
اگه بخوایم همه از یه کانفیگ باشن چی؟

**Reply by Hazily1536** [reply]
روش پیشنهادی دوم حذف شد؟ یادمه ۲تا دومین می خواست!

**Reply by lymanjre** [reply]
> روش پیشنهادی دوم حذف شد؟ یادمه ۲تا دومین می خواست!

نه هست

**Reply by lymanjre** [reply]
> # روش پیشنهادی دوم:
> 
> در این روش به ۲ دامنه نیاز داریم که دامنه اول را طبق روش قبل استفاده می کنیم یعنی چندین ساب دامین CDN مختلف برای پنل خود در کلادفلر تعریف کنید. دفت کنید که باید تیک پروکسی فعال باشد. [نحوه تعریف دامنه](https://github.com/hiddify/hiddify-config/wiki/%D8%A7%D9%86%D9%88%D8%A7%D8%B9-%D8%AF%D8%A7%D9%85%D9%86%D9%87-%D9%88-%D9%86%D8%AD%D9%88%D9%87-%D8%AB%D8%A8%D8%AA-%E2%80%8C%D8%A2%D9%86%E2%80%8C%D9%87%D8%A7)
> 
> مثلا:
> 
> sub1.domain1.com (proxy on)
> 
> sub2.doamin1.com (proxy on)
> 
> sub3.domain1.com (proxy on)
> 
> بعد برای آیپی های تمیز هم به همان روش بالا جلو برید. مثلا:
> 
> mci.domain1.com (proxy off) ==> آیپی تمیز کلادفلر در شبکه همراه اول
> 
> irc.domain1.com (proxy off) ==> آیپی تمیز کلادفلر در شبکه ایرانسل
> 
> mkb.domain1.com (proxy off) ==> آیپی تمیز کلادفلر در شبکه مخابرات
> 
> در نهایت اینجا یه تفاوت داریم با روش قبل. میایم با استفاده از دامین دوم به ازاء هر رکورد از دامین اول یک رکورد cname تعریف می کنیم. این کار برای جلوگیری از لو رفتن و فیلتر شدن دامنه و ساب دامنه های اول می باشد. یعنی مثلا:
> 
> sub1.domain2.com ==> sub1.domain1.com (proxy off)
> 
> sub2.domain2.com ==> sub2.domain1.com (proxy off)
> 
> sub3.domain2.com ==> sub3.domain1.com (proxy off)
> 
> mci.domain2.com ==> mci.domain1.com (proxy off)
> 
> irc.domain2.com ==> irc.domain1.com (proxy off)
> 
> mkb.domain2.com ==> mkb.domain1.com (proxy off)
> 
> در نهایت در پنل هیدیفای به منوی دامنه‌ها بروید. برای هر کدام از رکوردهای ثبت شده یک دامنه ایجاد کنید.
> 
> مثلا برای **همراه اول**
> 
> Domain: sub1.domain2.com
> 
> Type: CDN
> 
> Force CDN host: mci.domain2.com
> 
> یا برای **ایرانسل**
> 
> Domain: sub2.domain2.com
> 
> Type: CDN
> 
> Force CDN host: irc.domain2.com
> 
> یا برای **شبکه مخابرات**
> 
> Domain: sub3.domain2.com
> 
> Type: CDN
> 
> Force CDN host: mkb.domain2.com
> 
> **نکته بسیار مهم:**
> 
>     * در فیلد نمایش کانفیگ های دامنه می توانید دامنه های CDN دیگر را از لیست حذف نمایید. در این حالت کانفیگ های همین دامنه فقط نمایش داده می شود . باعث می شود آیپی تمیز ثبت شده به درستی کار کند.
> 
> 
> **و توجه داشته باشید:**
> 
>     * تمام این روشها راه حل های ابتکاری هستند و شما می توانید با دانش و خلاقیت خودتون تغییر بدین.

**روش پیشنهادی سوم:**

برای این حالت می‌توانید از قابلیت دامنه AutoCDN استفاده نمایید.
یعنی یک ساب دامین با پروکسی روشن در کلافلر بسازید. بدین صورت:
sub.domain.com (proxy on)
سپس حالت دامنه را روی AutoCDN قرار دهید.
در فیلد اجبار به استفاده از آیپی‌های زیر از[ الگوی پیشنهادی](https://ip-format.hiddify.com/) تگ‌های مربوط به شبکه‌های مورد نظر خود را وارد کنید. مثلا:
mci.ircf.space MCI 
mtn.ircf.space MTN 
mkh.ircf.space MKH 
rtl.ircf.space RTL 
و سپس تنظیمات را ذخیره نمایید. برای درک بهتر نحوه استفاده از این الگو  این مطلب را مطالعه کنید.

در نهایت شما با این روش یک دامنه AutoCDN برای استفاده روی چند شبکه مختلف خواهید داشت.

**Comment by yakoza021** [comment]
عالییییی

**Comment by MehdiZakhire** [comment]


**Reply by lymanjre** [reply]
داداش پیشنهاد میدم حتما یه بار با دقت بخونی. اولا که ادمین درگیر میشه نه کاربر که خوبی این روش همینه. دوم اینکه اگه از یه ساب واسه همه استفاده کنی احتمال قعطی هست بخاطر راندرابین dns

**Reply by alisadeghi1997** [reply]
سلام ای پی تمیز هارو تو کجای ساب دامین ها ست کنیم؟

**Reply by lymanjre** [reply]
توی رکوردهای mci,irc,… به بار از بالا کامل بخون

**Reply by alisadeghi1997** [reply]
منظورم این بود این آی پی تمیز ها باید روی ساب دامین پروکسی خاموش داخل کلود فلر ست بشن یا ساب دامین پروکسی خاموش داخل پنل؟
